src/transport.rs

```diff
@@ -221,7 +221,8 @@ impl Transport {
             Transport::Tcp { .. } => (),
             #[cfg(feature = "tls")]
             Transport::EncryptedTcp { .. } => (),
-            _ => panic!("connection streaming is only supported over TCP"),
+            #[cfg(feature = "unix-socket")]
+            Transport::Unix { .. } => panic!("connection streaming is only supported over TCP"),
```

Comment:
This change prevents an "unreachable branch" warning on builds without unix-socket support (e.g. on windows)